tuning/utils/config_utils.py

```diff
@@ -135,3 +135,32 @@ def txt_to_obj(txt):
     except UnicodeDecodeError:
         # Otherwise the bytes are a pickled python dictionary
         return pickle.loads(message_bytes)
+
+
+def get_last_checkpoint(train_args):
```

Comment:
why is this logic needed? since it seems to be [duplicated from that inside HF trainer](https://github.com/huggingface/transformers/blob/8820fe8b8c4b9da94cf1e4761876f85c562e0efe/src/transformers/trainer.py#L1930)

Reply:
This is also doing similar work to what exists in [build/utils](https://github.com/foundation-model-stack/fms-hf-tuning/blob/main/build/utils.py#L43-L63) so that or this method could be refactored to be used in both places

Reply:
That's correct. I missed that one. Now importing it from `transformers.trainer_utils import get_last_checkpoint`

Reply:
I think it would be good to reuse this flag to pass to train() function as is the way user provides, this would save us from too much of custom handling which limit other potential usecases.

https://github.com/huggingface/transformers/blob/8820fe8b8c4b9da94cf1e4761876f85c562e0efe/src/transformers/training_args.py#L663

Reply:
Just to get clarity @kmehant, does user passes the flag `resume_from_checkpoint` if users wants to resume from particular checkpoint ?
And in particular case where user doesn't pass the flag, do we assume that user doesn't intend on resuming the tuning and wants to start a new one, OR we still do need to check for latest checkpoint path and use it to resume tuning. 
Thanks!

Reply:
> Just to get clarity @kmehant, does user passes the flag resume_from_checkpoint if users wants to resume from particular checkpoint ?

Yes, `--resume_from_checkpoint=true` if to start from last checkpoint or to a path if it has to be some arbitrary checkpoint. And then we pass this as is to the train()

> And in particular case where user doesn't pass the flag, do we assume that user doesn't intend on resuming the tuning and wants to start a new one, OR we still do need to check for latest checkpoint path and use it to resume tuning.

I have no particular suggestion on the default flow when user does not pass the flag 🙂

Reply:
>Yes, --resume_from_checkpoint=true if to start from last checkpoint or to a path if it has to be some arbitrary checkpoint. And then we pass this as is to the train()

Okay